githubapp/client_creator.go

```diff
@@ -229,6 +266,72 @@ func applyMiddleware(base *http.Client, middleware []ClientMiddleware) {
 	}
 }
 
+func newAppInstallation(integrationID int, privKeyBytes []byte, v3BaseURL string) (ClientMiddleware, error) {
+	var transportError error
+	installation := func(next http.RoundTripper) http.RoundTripper {
+		itr, err := ghinstallation.NewAppsTransport(next, integrationID, privKeyBytes)
+		if err != nil {
+			transportError = err
+			return next
+		}
+		// leaving the v3 URL since this is used to refresh the token, not make queries
+		itr.BaseURL = strings.TrimSuffix(v3BaseURL, "/")
+		return itr
+	}
+	if transportError != nil {
```

Comment:
This will always be `nil` at this point, because it can only be set once the `installation` middleware in called. To use the closure-error approach, we have to check the error after constructing the client and invoking all the middleware.